src/fighter_state.rs

```diff
@@ -955,9 +963,19 @@ fn dying(
 /// Throw the item in the player's inventory
 fn throwing(
     mut commands: Commands,
-    mut fighters: Query<(Entity, &Transform, &Facing, &mut Inventory), With<Throwing>>,
+    mut fighters: Query<
+        (
+            Entity,
+            &Transform,
+            &Facing,
+            &mut Inventory,
+            Option<&Children>,
+        ),
+        With<Throwing>,
+    >,
+    being_hold: Query<Entity, With<BeingHold>>,
```

Comment:
I feel it might make more sense to add an entirely new player state, something like `holding` and its transition system `transition_from_holding` with a higher priority than attacking, which I think is generally how we intend to structure our state model going forward, and would remove the need for checking if we are holding in the attack system as well since the higher priority state would simply take precedent when the transitions were determined.

Reply:
I implemented the states but not sure if it's how you wanted, or if it's a good implementation.